src/rewards/voters.rs

```diff
@@ -33,14 +37,65 @@ pub fn calculate_stake<'address>(
 pub fn calculate_reward_share<'address>(
     total_stake: u64,
     stake_per_voter: &HashMap<&'address Address, u64>,
+    threshold_addresses: &AddressesVoteCount,
+    threshold: u64,
 ) -> HashMap<&'address Address, U64F64> {
     stake_per_voter
         .iter()
-        .map(|(k, v)| (*k, fixed::types::U64F64::from_num(*v) / total_stake as u128))
+        .map(|(k, v)| {
+            let reward = if *threshold_addresses
+                .get(k)
+                .unwrap_or_else(|| panic!("Missing address {} in vote count map", k))
+                >= threshold
+            {
+                U64F64::from_num(*v) / total_stake as u128
+            } else {
+                U64F64::default()
```

Comment:
nit: `U64F64::ZERO` would be easier to read IMO

Reply:
Yep, and I wanted to do that. But the hint was missing from my IDE and I took the easy path 😅